src/GitVersion.Core/VersionCalculation/ContinuousDeliveryVersionCalculator.cs

```diff
@@ -0,0 +1,57 @@
+using System.Diagnostics.Contracts;
+using GitVersion.Common;
+using GitVersion.Logging;
+
+namespace GitVersion.VersionCalculation;
+
+internal sealed class ContinuousDeliveryVersionCalculator : NonTrunkBasedVersionCalculatorBase, IContinuousDeliveryVersionCalculator
+{
+    public ContinuousDeliveryVersionCalculator(ILog log, IRepositoryStore repositoryStore, Lazy<GitVersionContext> versionContext)
+        : base(log, repositoryStore, versionContext)
+    {
+    }
+
+    public SemanticVersion Calculate(NextVersion nextVersion)
+    {
+        using (this.log.IndentLog("Using continues delivery typology to calculate the incremented version!!"))
+        {
+            var preReleaseTag = nextVersion.IncrementedVersion.PreReleaseTag;
+            if (!preReleaseTag.HasTag() || !preReleaseTag.Number.HasValue)
+                throw new WarningException("--PRE--CONDITION--FAILED--");
```

Comment:
Why the strange warning message `--PRE--CONDITION--FAILED—`? I like that exceptions convey a user-friendly, descriptive error message. Can we say something like this instead?

```suggestion
                throw new WarningException("No pre-release tag found");
```

Reply:
Done